recipes/cassandra-cpp-driver/all/conanfile.py

```diff
@@ -0,0 +1,116 @@
+import os.path
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+
+
+class CassandraCppDriverConan(ConanFile):
+    name = "cassandra-cpp-driver"
+    license = "Apache-2.0"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://docs.datastax.com/en/developer/cpp-driver/"
+    description = "DataStax C/C++ Driver for Apache Cassandra and DataStax Products"
+    topics = ("cassandra", "cpp-driver", "database", "conan-recipe")
+
+    settings = "os", "compiler", "build_type", "arch"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "install_header_in_subdir": [True, False],
+        "use_atomic": [None, "boost", "std"],
+        "with_openssl": [True, False],
+        "with_zlib": [True, False],
+        "with_kerberos": [True, False],
+        "use_timerfd": [True, False]
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "install_header_in_subdir": False,
+        "use_atomic": None,
+        "with_openssl": True,
+        "with_zlib": True,
+        "with_kerberos": False,
+        "use_timerfd": True
+    }
+
+    generators = "cmake"
+    exports_sources = [
+        "CMakeLists.txt",
+        "patches/*"
+    ]
+
+    def _configure_cmake(self):
+        cmake = CMake(self)
+        cmake.definitions["VERSION"] = self.version
+        cmake.definitions["CASS_INSTALL_HEADER_IN_SUBDIR"] = self.options.install_header_in_subdir
+
+        if self.settings.os == "Macos":
+            raise ConanInvalidConfiguration("Macos is unsuported")
+
+        if self.options.use_atomic == "boost":
+            # Compilation error on Linux with clang
+            # cmake.definitions["CASS_USE_BOOST_ATOMIC"] = True
+            # cmake.definitions["CASS_USE_STD_ATOMIC"] = False
+            raise ConanInvalidConfiguration(
+                "Boost.Atomic is not supported at the moment")
```

Comment:
Does it work for gcc@Linux?
If so, you can move the throw to `configure` (and only if `use_atomic=True` and `compiler=clang`) and set the cmake definitions here unconditionally.

Reply:
I don't think. Bincrafters has a similar recipe, Linux is excluded: https://github.com/bincrafters/conan-cassandra-driver

Reply:
Tested with GCC and Clang on Linux - compilation failed. Boost.Atomic is switched off for Linux. But I switched it on for Windows.
[Done at](https://github.com/conan-io/conan-center-index/pull/3110/commits/039575049d0ee4d84115fb5025296479aa4577dc)